packages/grid/x-data-grid-premium/src/DataGridPremium/DataGridPremium.tsx

```diff
@@ -35,7 +35,7 @@ const DataGridPremiumRaw = React.forwardRef(function DataGridPremium<R extends G
     <GridContextProvider privateApiRef={privateApiRef} props={props}>
       <GridRoot
         className={props.className}
-        style={props.style}
+        style={{ cursor: 'default', ...props.style }}
```

Comment:
I think this is not gonna work as it will override the `<GridRoot />` cursor instead of `<GridCell  />`'s without taking into consideration the condition based on which it should be changed.

Alternatively, you could explore adding a class to cell classes like [this one](https://github.com/mui/mui-x/blob/master/packages/grid/x-data-grid/src/components/containers/GridRootStyles.ts#L385) and use it in the cell classes based on the `unstable_cellSelection` flag [here](https://github.com/mui/mui-x/blob/master/packages/grid/x-data-grid/src/components/cell/GridCell.tsx#L110).